ObjectiveGit/GTRepository.m

```diff
@@ -718,24 +718,33 @@ - (GTSubmodule *)submoduleWithName:(NSString *)name error:(NSError **)error {
 
 #pragma mark User
 
++ (NSString *)defaultUserName {
+	NSString *name = NSFullUserName();
+	if (name.length == 0) name = NSUserName();
+	if (name.length == 0) name = @"nobody";
+	return name;
+}
+
++ (NSString *)defaultEmail {
+	NSString *username = NSUserName();
+	if (username.length == 0) username = @"nobody";
+	NSString *domain = NSProcessInfo.processInfo.hostName ?: @"nowhere.local";
+	return [NSString stringWithFormat:@"%@@%@", username, domain];
+}
+
 - (GTSignature *)userSignatureForNow {
 	GTConfiguration *configuration = [self configurationWithError:NULL];
 	NSString *name = [configuration stringForKey:@"user.name"];
-	if (name.length == 0) {
-		name = NSFullUserName();
-		if (name.length == 0) name = NSUserName();
-		if (name.length == 0) name = @"nobody";
-	}
+	if (name.length == 0) name = self.class.defaultUserName;
 
 	NSString *email = [configuration stringForKey:@"user.email"];
-	if (email == nil) {
-		NSString *username = NSUserName();
-		if (username.length == 0) username = @"nobody";
-		NSString *domain = NSProcessInfo.processInfo.hostName ?: @"nowhere.local";
-		email = [NSString stringWithFormat:@"%@@%@", username, domain];
-	}
+	if (email == nil) email = self.class.defaultEmail;
```

Comment:
Libgit2 used to be fine with empty emails: https://github.com/libgit2/libgit2/commit/76e3c43fb99eb75cf33affb1a93260af4d472da5

Apparently not anymore :cry: